front/src/components/topMenu/ClientMapModal.js

```diff
@@ -0,0 +1,144 @@
+  const warehouseClient = new WarehouseClient(auth);
+  const defaultMapState = { center: [55.76, 37.64], zoom: 10, controls: [] };
+  const [mapState, setMapState] = useState(defaultMapState);
+  const [errorMsg, setErrorMsg] = useState();
+  const [yaMaps, setYaMaps] = useState();
+  const searchRef = useRef(null);
+  const [map, setMap] = useState();
+
+  const onYmapsLoad = ymaps => {
+    setYaMaps(ymaps);
+    new ymaps.SuggestView(searchRef.current);
+  }
+
+  const onAddressSelect = async () => {
+    const fullAddress = searchRef.current.value;
+    console.log(fullAddress);
+    console.log(yaMaps);
+    const result = await yaMaps.geocode(fullAddress);
+        let geoObject = result.geoObjects.get(0),
+                  error,
+                  hint;
+        if (geoObject) {
+          const preciseProperty = 'metaDataProperty.GeocoderMetaData.precision';
+          switch (geoObject.properties.get(preciseProperty)) {
+            case 'exact':
+                  break;
+            case 'number':
+            case 'near':
+            case 'range':
+            case 'street':
+                  error = 'Неполный адрес, требуется уточнение';
+                  hint = 'Уточните номер дома';
+                  break;
+            case 'other':
+            default:
+                  error = 'Неточный адрес, требуется уточнение';
+                  hint = 'Уточните адрес';
+            }
+        } else {
+          error = 'Адрес не найден';
+          hint = 'Уточните адрес';
+        }
+        if (error) {
+          setErrorMsg(error+"\n"+hint);
+        } else {
+          const coord = geoObject.geometry.getCoordinates();
+          const shortAddress = geoObject.properties.get('name');
+          const coords = { 'lat' : coord[0], 'lon' : coord[1]};
+          const response = await warehouseClient.getNearestWarehouse(coords);
+          if (response && response.success) {
+            console.log(response.data);
+            const warehouseId = response.data.id;
+            setAddress({fullAddress, shortAddress, coord, warehouseId});
+            setMapState({center: coord, zoom: 17, controls: []});
+            setErrorMsg(null);
+          } else {
+            setErrorMsg("Доставка в указанную зону недоступна.");
+            setMapState(defaultMapState);
+            setAddress({});
+          }
+        }
+    }
+
+    return (
+      <Modal
+             width={700}
+             centered={true}
+             footer={null}
+             title={
+               address && Object.keys(address).length == 0
+               ? "Select delivery address"
+               : "Edit delivery address"
+             }
+             visible={isMapVisible || (auth.token && address && Object.keys(address).length == 0)}
+             okText="Choose"
+             onCancel = {() => {setIsMapVisible(false);
+                                searchRef.current.value = address.fullAddress;}
```

Comment:
```suggestion
                                if(address.fullAddress) searchRef.current.value = address.fullAddress;
                                else setErrorMsg("Необходимо ввести и подтвердить адрес");}
```
Если закрыть модальное окно, но не вводить или вводить и не подтверждать адрес, в placeholder (searchRef здесь) записывается undefined и юзер это видит. Как вариант, я предложил такое решение